
syntax = "proto3";

package Lya.ProtocolBuffers;

service LyaService {
    rpc sync(PBSyncRequest) returns (PBSyncResponse);
    rpc check_availability(PBAvailabilityRequest) returns (PBAvailabilityResponse);
}

message PBAvailabilityRequest {
}

message PBAvailabilityResponse {
}

message PBSyncRequest {
    repeated string functions = 1;
    repeated string files = 2;
}

message PBSyncResponse {
    repeated PBFileToLocalizations file_to_localizations = 1;
}

message PBFileToLocalizations {
    string file = 1;
    repeated PBLocalization localizations = 2;
}

message PBLocalization {
    string id = 1;
    string text = 2;
    repeated string params = 3;
    int64 line = 4;
    int64 column = 5;
}
